### DISCUSSION TITLE
Immer and react-hook-form - help with unfreezing values

### DISCUSSION THREAD

**Discussion by wdfinch** [react-hook-form/react-hook-form#4502] [Q&A]
Currently, react-hook-form does not work with `immer` when object freezing is enabled. My two workarounds are:

1. Use the `setAutoFreeze` to disable auto freezing in `immer`.
2. Perform a custom deep clone of the object and remove freezing.

I would like to do 2 so I don't have to disable the safety that auto-freezing provides me. The issue I can't figure out how to perform the clone to remove the freezing. I've attached a sandbox below. I'm truly stumped on why this doesn't work. I create a new array and a new object entirely. To take it even further, I'm not even using the string I would want to clone. I'm using a new string entirely. Can anyone explain why this is still not working? I would really appreciate it.

So in other words, I would like the function on line 22 to produce a new array that the library will accept and not crash. 

https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-l6l9l?file=/src/index.js

**Comment by wdfinch** [comment]
I figured this out.
